README.rst

```diff
@@ -184,6 +199,125 @@ it should be easier to use in most cases.
+(i.e. to the URL of a website you want to render), not to the URL of the
+requested Splash endpoint. "True" URL is still available as
+``response.real_url``.
+
+SplashJsonResponse provide extra features:
+
+* ``response.data`` attribute contains response data decoded from JSON;
+  you can access it like ``response.data['html']``.
+
+* If Splash session handling is configured, you can access current cookies
+  as ``response.cookiejar``; it is a CookieJar instance.
+
+* If Scrapy-Splash response magic is enabled in request (default),
+  several response attributes (headers, body, url, status code)
+  are set automatically from original response body:
+
+  * response.headers are filled from 'headers' keys;
+  * response.url is set to the value of 'url' key;
+  * response.body is set to the value of 'html' key,
+    or to base64-decoded value of 'body' key;
+  * response.status is set from the value of 'http_status' key.
+
+When ``respone.body`` is updated in SplashJsonResponse
+(either from 'html' or from 'body' keys) familiar ``response.css``
+and ``response.xpath`` methods are available.
+
+To turn off special handling of JSON result keys either set
+``meta['splash']['magic_response']=False`` or pass ``magic_response=False``
+argument to SplashRequest.
+
+Session Handling
+================
+
+Splash itself is stateless - each request starts from a clean state.
+In order to support sessions the following is required:
+
+1. client (Scrapy) must send current cookies to Splash;
+2. Splash script should make requests using these cookies and update
+   them from HTTP response headers or JavaScript code;
+3. updated cookies should be sent back to the client;
+4. client should merge current cookies wiht the updated cookies.
+
+For (2) and (3) Splash provides ``spalsh:get_cookies()`` and
+``splash:init_cookies()`` methods which can be used in Splash Lua scripts.
+
+ScrapyJS provides helpers for (1) and (4): to send current cookies
+in 'cookies' field and merge cookies back from 'cookies' response field
+set ``request.meta['splash']['args']['session_id']`` to the session
+identifier. If you only want a single session use the same ``session_id`` for
+all request; any value like '1' or 'foo' is fine.
+
+For ScrapyJS session handling to work you must use ``/execute`` endpoint
+and a Lua script which accepts 'cookies' argument and returns 'cookies'
+field in the result::
+
+   function main(splash)
+       splash:init_cookies(splash.args.cookies)
+
+       -- ... your script
+
+       return {
+           cookies = splash:get_cookies(),
+           -- ... other results, e.g. html
+       }
+   end
+
+SplashRequest sets ``session_id`` automatically for ``/execute`` endpoint,
+i.e. cookie handling is enabled by default if you use SplashRequest,
+``/execute`` endpoint and a compatible Lua rendering script.
+
+If you want to start from the same set of cookies, but then 'fork' sessions
+set ``request.meta['splash']['args']['new_session_id']`` in addition to
```

Comment:
A good catch; the code is correct. `args` are arguments sent to Splash; `new_session_id` is a SplashMiddleware option.